### PR TITLE
Update logger from gokit to base/log

### PR DESCRIPTION
**Pull request by vxio**
Let me know if you'd like me to change the log messages or configuration in any areas; I tried to maintain the original log messages in this PR.

### CODE REVIEW

File: x/route/route.go [hunk 1]
```diff
@@ -111,6 +98,19 @@ func wrapResponseWriter(logger log.Logger, w http.ResponseWriter, r *http.Reques
 	return ww, nil
 }
 
+// todo: temporary adapter until moovhttp.Wrap() gets updated to use base/log
+type loggerAdapter struct {
```
Comment (vxio) [inline x/route/route.go:102]: New issue: https://github.com/moov-io/base/issues/117

File: x/trace/trace.go [hunk 2]
```diff
@@ -77,3 +74,18 @@ func setupTracer(logger log.Logger, cfg jaegercfg.Configuration) (opentracing.Tr
 func GlobalTracer() opentracing.Tracer {
 	return opentracing.GlobalTracer()
 }
+
+var _ jaeger.Logger = (*jaegerLogger)(nil)
+
+// adapter for jaeger.Logger
+type jaegerLogger struct {
```
Comment (vxio) [inline x/trace/trace.go:81]: Jaeger comes with a go-kit adapter: "github.com/jaegertracing/jaeger-lib/client/log/go-kit"
But we need to write our own here.

File: pkg/transfers/inbound/returns.go [hunk 3]
```diff
@@ -89,7 +93,7 @@ func (pc *returnProcessor) processReturnEntry(fh ach.FileHeader, bh *ach.BatchHe
 	// Do we find a Transfer related to the ach.EntryDetail?
 	transfer, err := pc.transferRepo.LookupTransferFromReturn(amount, entry.TraceNumber, effectiveEntryDate)
 	if transfer != nil {
-		pc.logger.Log("inbound", fmt.Sprintf("handling return for transferID=%s", transfer.TransferID))
+		pc.logger.Set("transferID", transfer.TransferID).Log("handling return for transfer")
```
Comment (alovak) [inline pkg/transfers/inbound/returns.go:96]: I would create new logger `logger := pc.logger.Set("transferID", transfer.TransferID)` and use it in line 115 (with more context).

File: pkg/transfers/router.go [hunk 4]
```diff
@@ -57,11 +57,10 @@ func NewRouter(
...
-		err = fmt.Errorf("problem creating transfer limiter: %v", err)
-		cfg.Logger.Log("transfers", err)
-		panic(err.Error())
+		err = cfg.Logger.LogErrorf("problem creating transfer limiter: %v", err).Err()
+		panic(err)
 	}
-	cfg.Logger.Log("transfers", fmt.Sprintf("setup %T limit checker", limitChecker))
+	cfg.Logger.Logf("setup %T limit checker", limitChecker)
 	return &Router{
 		Logger:    cfg.Logger,
```
Comment (alovak) [inline pkg/transfers/router.go:65]: what do you think about adding more context to logger like this:
```go
Logger:    cfg.Logger.Set("package", "transfers"),
```

Comment (alovak) [reply pkg/transfers/router.go:65]: I see it's useless as passed instance of logger will not be use inside router (at least I don't see where it's used).

File: x/trace/trace.go (hunk 2 above)
Comment (vxio) [reply x/trace/trace.go:81]: Maybe I should move this to `base/logger` eventually?

**Review by alovak — APPROVED** [review]

**Comment by adamdecaf** [comment]
Thanks for grabbing this.

**Pull request merged**